lib/hardwareWalletUtilities.js

```diff
@@ -0,0 +1,106 @@
+import { Script, Outpoint, TX, MTX, Coin } from 'bcoin';
+import { LedgerBcoin, U2F, LedgerTXInput } from 'bledger/lib/bledger-browser';
+const { Device } = U2F;
+
+/**
+ * Initializes and returns a new Device
+ * NOTE: caller must call device.close when done
+ * @returns {Device}
+ */
+const newDevice = async () => {
+  try {
+    const devices = await Device.getDevices();
+    const device = new Device({
+      device: devices[0],
+      timeout: 100000
+    });
+    return device;
+  } catch (e) {
+    throw e;
+  }
+}
+
+/**
+ * Initializes and returns a new LedgerBcoin
+ * @returns {LedgerBcoin}
+ */
+export const newLedgerBcoin = async () => {
+  try {
+    const device = await newDevice();
+    const ledgerBcoin = new LedgerBcoin({ device });
+
+    // return device so that it can be closed
+    return { device, ledgerBcoin };
+
+  } catch (e) {
+    console.error(e);
+    return null;
+  }
+}
+
+/**
+ * Get public key from an application by path
+ * @param {btcApp} - an instance of a key manager like LedgerBcoin
+ * @param {path} - bip 44 path
+ */
+export const getPublicKey = async (btcApp, path) => {
+  try {
+    return await btcApp.getPublicKey(path);
+  } catch (e) {
+    console.error(e);
+    return null;
+  }
+}
+
+/**
+ * Get BIP 44 path by account
+ * @param {account} - a BIP 44 account integer
+ * @returns {path} - bip 44 path
+ */
+// TODO: be flexible for testnet ( m/44'/1'/0 )
+export const buildXpubPath = account => `m/44'/0'/${account}'`;
+
+// TODO: clean up arguments
+export const signTX = async (bcoinApp, client, accountPath, proposalMTX, walletId) => {
```

Comment:
I think (but I'm not sure) Nodar has an update that might help take care of some of this.